lua/entities/gmod_tardis/modules/sh_redecorate.lua

```diff
@@ -122,13 +151,14 @@ if SERVER then
 
         self:ForceDematState()
 
-        local ext_saved_data = self:GetData("redecorate_parent_ext_data")
-        if ext_saved_data then
-            for name,value in pairs(ext_saved_data) do
+        local saved_data = self:GetData("redecorate_parent_data")
+        if saved_data then
+            for name,value in pairs(saved_data) do
                 self:SetData(name, value, true)
             end
         end
-        self:SetData("redecorate_parent_ext_data", nil, true)
+        self:SetPhyslock(saved_data["physlock"])
```

Comment:
Should this be handled here? It seems like this should maybe go into the physlock module as part of the MigrateData hook